### DISCUSSION TITLE
GCC on MacOS

### DISCUSSION THREAD

**Discussion by alexandru-andronache** [microsoft/vscode-cpptools#9010] [Q&A] [answered]
Hi,

What is the reason for GCC not being supported on mac os? I don't see it in the list of the compilers and platforms and when i try to use it even for a simple hello world program it will fail with linker errors:

`[build] Undefined symbols for architecture x86_64:
[build]   "_std::ostream::operator<<(std::ostream& (*)(std::ostream&))", referenced from:
[build]       _main in main.cpp.o
[build]   "_std::ostream::operator<<(int)", referenced from:
[build]       _main in main.cpp.o
[build]   "_std::ios_base::Init::Init()", referenced from:
[build]       ___static_initialization_and_destruction_0(int, int) in main.cpp.o
[build]   "_std::ios_base::Init::~Init()", referenced from:
[build]       ___static_initialization_and_destruction_0(int, int) in main.cpp.o
[build]   "_std::cout", referenced from:
[build]       _main in main.cpp.o
[build]   "_std::basic_ostream<char, std::char_traits<char> >& std::endl<char, std::char_traits<char> >(std::basic_ostream<char, std::char_traits<char> >&)", referenced from:
[build]       _main in main.cpp.o
[build] ld: symbol(s) not found for architecture x86_64`

**Comment by sean-mcmanus** [comment, marked as answer]
It's possible to get gcc to work on Mac using Cellar or Brew. We don't provide any documentation/help on how to do that though. And we support using it with our extension, but it sounds like you're hitting build issues unrelated to our extension.

**Reply by alexandru-andronache** [reply]
I've installed gcc using brew. VS Code detects it, i can see it in the list of the compilers, but when i try to use it i get linker errors. GCC seems to be working since i can use it to compile a simple cpp file but when i try to use GCC from VS code it will not work. 
If you say that this should work i will investigate more to see what is the problem.

**Reply by sean-mcmanus** [reply]
Yeah, looks like there's some problem linking to the libstdc++ ostream library. It's working for me -- seems to load /usr/local/Cellar/gcc/10.2.0_2/lib/gcc/10/libstdc++.6.dylib (when debugging).
